src/mainwindow.h

```diff
@@ -109,6 +109,8 @@ class MainWindow : public QMainWindow
     void openFailed(QString);
     void aboutToShutDown();
     void renameRequested();
+    void serviceInChanged(int delta, Mlt::Service*);
```

Comment:
I am not sure about the location of this. But MainWindow has access to all the panels that need to be connected to this event.

Reply:
It is true that MainWindow is like the main controller and a patch bay for many signals. However, it can be preferable to make each component emit its own signal, and in MainWindow connect each of those. This is what it currently does for service in/out changed, but you added emissions to Mlt::Controller. Mlt::Controller is not a QObject and lacks signals, but GLWidget inherits it and adds that. However, if you add this signal to GLWidget you cannot emit it from Mlt::Controller. So, I think this stays the way you have it.